rbac/api/cross_access/view.py

```diff
@@ -159,6 +216,21 @@ def throw_validation_error(self, source, message):
         error = {source: [message]}
         raise ValidationError(error)
 
+    def update_cross_account_request(self, car, update_data):
+        """Update cross-account-request with list of role names."""
+        if car.status == "pending" and update_data.get("status").lower() == "approved":
+            # create_principal_for_approved_request()
+            pass
+        for field in update_data:
```

Comment:
Should we maybe wrap this in a transaction in case any of the updates fail, we won't wipe out the roles first and commit that?